src/Camille.RiotGames/src/Util/RegionalRequester.cs

```diff
@@ -67,6 +92,18 @@ public RegionalRequester(IRiotGamesApiConfig config, string route)
             var retries = 0;
             var num429s = 0;
             var num5xxs = 0;
+
+            // If the region is not being used as a subdomain
+            if (!_regionAsSubdomain)
+            {
+                // Append the region as a query parameter, with consideration for other parameters
+                var uri = request.RequestUri.ToString()
+                          + (!request.RequestUri.ToString().Contains("?") ? "?" : "&")
+                          + $"region={_region}";
+                // Replace the request with a new one that has the region correctly appended
+                request = new HttpRequestMessage(request.Method, uri);
+            }
```

Comment:
Should be able to eliminate this extra logic if `regionAsSubdomain` is in the configspec

Reply:
Unfortunately: no.
The bulk of the logic around line 101 in your snippet (now [line 105](https://github.com/MingweiSamuel/Camille/blob/f3906c7d8895415792ce48194bf7990a6e5eaa2c/src/Camille.RiotGames/src/Util/RegionalRequester.cs#L105)) is instead based off of whether the endpoint uses query parameters based on the method's optional parameters - but since those parameters are baked into the URL instead of specified as `request.Properties` (I think?) I don't see how this could get much cleaner.